test-resources/src/functions/common/dynamoDBAdapter/dynamoDBAdapter.ts

```diff
@@ -0,0 +1,65 @@
+import { DynamoDBClient, PutItemCommand } from "@aws-sdk/client-dynamodb";
+import { marshall } from "@aws-sdk/util-dynamodb";
+import { NodeHttpHandler } from "@smithy/node-http-handler";
+import { PutItemOperation } from "../../common/dynamoDBAdapter/putItemOperation";
+import { emptyFailure, emptySuccess, Result } from "../../common/utils/result";
+import { LogMessage } from "../logging/LogMessage";
+import { logger } from "../logging/logger";
+
+export class DynamoDBAdapter {
```

Comment:
nit: it would be good to be explicit here about this implements an interface
```suggestion
export class DynamoDBAdapter implements ICredentialResultRegistry {
```

Reply:
Updated